evalml/tests/automl_tests/test_automl_iterative_algorithm.py

```diff
@@ -0,0 +1,1146 @@
+        total = 1 + len(automl.allowed_pipelines)
+        total += (num_batches - 1) * batch_size
+        return total
+
+    X, y = X_y_binary
+
+    # Checking for default of _pipelines_per_batch
+    automl = AutoMLSearch(
+        X_train=X,
+        y_train=y,
+        problem_type="binary",
+        max_batches=2,
+        optimize_thresholds=False,
+        _automl_algorithm="iterative",
+    )
+    env = AutoMLTestEnv("binary")
+    with env.test_context(score_return_value={"Log Loss Binary": 0.30}):
+        automl.search()
+    assert automl._pipelines_per_batch == 5
+    assert automl._automl_algorithm.pipelines_per_batch == 5
+    assert total_pipelines(automl, 2, 5) == len(automl.full_rankings)
+
+    automl = AutoMLSearch(
+        X_train=X,
+        y_train=y,
+        problem_type="binary",
+        max_batches=1,
+        optimize_thresholds=False,
+        _pipelines_per_batch=2,
+        _automl_algorithm="iterative",
+    )
+    with env.test_context(score_return_value={"Log Loss Binary": 0.30}):
+        automl.search()
+    assert automl._pipelines_per_batch == 2
+    assert automl._automl_algorithm.pipelines_per_batch == 2
+    assert total_pipelines(automl, 1, 2) == len(automl.full_rankings)
+
+    automl = AutoMLSearch(
+        X_train=X,
+        y_train=y,
+        problem_type="binary",
+        max_batches=2,
+        optimize_thresholds=False,
+        _pipelines_per_batch=10,
+        _automl_algorithm="iterative",
+    )
+    with env.test_context(score_return_value={"Log Loss Binary": 0.30}):
+        automl.search()
+    assert automl._pipelines_per_batch == 10
+    assert automl._automl_algorithm.pipelines_per_batch == 10
+    assert total_pipelines(automl, 2, 10) == len(automl.full_rankings)
+
+
+def test_automl_respects_random_seed_iterative(
+    X_y_binary, dummy_classifier_estimator_class
+):
+    X, y = X_y_binary
+    automl = AutoMLSearch(
+        X_train=X,
+        y_train=y,
+        problem_type="binary",
+        allowed_component_graphs={"Name_0": [dummy_classifier_estimator_class]},
+        optimize_thresholds=False,
+        random_seed=42,
+        max_iterations=10,
+        _automl_algorithm="iterative",
+    )
+    pipelines = [
+        BinaryClassificationPipeline(
+            component_graph=[dummy_classifier_estimator_class], random_seed=42
+        )
+    ]
+    automl._automl_algorithm = IterativeAlgorithm(
+        X=X,
+        y=y,
+        problem_type="binary",
+        tuner_class=SKOptTuner,
+        random_seed=42,
+        n_jobs=1,
+        number_features=X.shape[1],
+        pipelines_per_batch=5,
+        ensembling=False,
+        text_in_ensembling=False,
+        pipeline_params={},
+        custom_hyperparameters=None,
+    )
+    automl._automl_algorithm.allowed_pipelines = pipelines
+    assert automl.allowed_pipelines[0].random_seed == 42
+
+
+def test_automl_respects_pipeline_parameters_with_duplicate_components_iterative(
```

Comment:
A little unsure if this tests makes the cut for the mock algorithm. On one end it uses `allowed_component_graphs` which is `Iterative` specific. On the other, it tests algorithm agnostic behavior (whether pipeline_parameters make it to the pipelines). I'll add it to the list for now.